wgpu-hal/src/dx12/conv.rs

```diff
@@ -329,14 +329,14 @@ fn map_stencil_face(face: &wgt::StencilFaceState) -> d3d12::D3D12_DEPTH_STENCILO
 
 pub fn map_depth_stencil(ds: &wgt::DepthStencilState) -> d3d12::D3D12_DEPTH_STENCIL_DESC {
     d3d12::D3D12_DEPTH_STENCIL_DESC {
-        DepthEnable: if ds.is_depth_enabled() { 1 } else { 0 },
+        DepthEnable: ds.is_depth_enabled().into(),
```

Comment:
I'm pretty biased against into - I would either use as or do the explicit if.

Reply:
Why is that? 🤔 The opinion at [`clippy::bool_to_int_with_if`](https://rust-lang.github.io/rust-clippy/master/index.html#bool_to_int_with_if) is that `into` is actually very idiomatic. Are you worried about problems like a foot-gun here?

Whatever the case, I don't mind reverting; this would require an alternate strategy, though, like `#![allow(clippy::bool_to_int_with_if)]` on crates, since it's now a warn-by-default lint.

Reply:
I personally use `T::from(v)` more often than `.into()`

Reply:
I agree with the lint that the if statement is unnecessary, I don't like into because it hides the destination type from the source, so unless you have an ide, you don't know the type. From would be fine, as would an explicit as.

We use `as _` around a bit and we probably shouldn't be either, but at least you know it's a primative cast.

Reply:
Move to use `From` with 342b36e5.